cime/src/drivers/mct/main/cime_comp_mod.F90

```diff
@@ -2523,8 +2546,94 @@ subroutine cime_run()
        !----------------------------------------------------------
        !| GLC SETUP-SEND
        !----------------------------------------------------------
+       if (glc_present) then
+
+          if (ocn_c2_glc) then
+             call prep_glc_calc_o2x_gx(timer='CPL:glcprep_ocn2glc') !remap ocean fields to o2x_g at atmospheric timestep
+          end if
+          if (lnd_c2_glc) then
+             ! Note that l2x_gx is obtained from mapping the module variable l2gacc_lx
+             call prep_glc_calc_l2x_gx(fractions_lx, timer='CPL:glcprep_lnd2glc')
+                !...and then here is where Bill S. does the new in-coupler downscaling.  I changed this so it's downscaling
+                ! non-accumulated fields.  This means (in theory) that it is called more frequently, but then can be merged 
+                ! and averaged with the other stuff, below.
+             call prep_glc_mrg(infodata, fractions_gx, timer_mrg='CPL:glcprep_mrgx2g')
+          end if
+
+          if (glc_c2_ocn) then
+             call prep_glc_calculate_subshelf_boundary_fluxes! this outputs
+                                           !x2g_g/g2x_g, where latter is going
+                                           !to ocean, so should get remapped to
+                                           !ocean grid in prep_ocn_shelf_calc_g2x_ox
+             call prep_ocn_shelf_calc_g2x_ox(timer='CPL:glcpost_glcshelf2ocn')
+                                           !Map g2x_gx shelf fields that were updated above, to g2x_ox.
+                                           !Do this at intrinsic coupling
+                                           !frequency
+             call prep_ice_shelf_calc_g2x_ix(timer='CPL:glcpost_glcshelf2ice')
+                                           !Map g2x_gx shelf fields to g2x_ix.
+                                           !Do this at intrinsic coupling
+                                           !frequency.  This is perhaps an
+                                           !unnecessary place to put this
+                                           !call, since these fields aren't
+                                           !changing on the intrinsic
+                                           !timestep.  But I don't think it's
+                                           !unsafe to do it here.
+          endif
+
+          call prep_glc_accum(lnd_c2_glc, timer='CPL:glcprep_accum') !accum x2g_g fields here into x2g_gacc, along with l2gacc_lx
```

Comment:
@JeremyFyke , I am assuming that this call to `prep_glc_accum` needs to be here so that accumulation will occur for coupling fields from both LND and now OCN.  However, there already is a call to  `prep_glc_accum` from routine `cime_run_lnd_recv_post` in this file at ~line 3831.  I am guessing that calling `prep_glc_accum` from multiple places is a bad idea as it could cause over-accumulation.  My intution would be to delete the call to `prep_glc_accum` in `cime_run_lnd_recv_post` and keep it here.  But I'm not sure if that generates any other problems.  Do you happen to have recollection of this or thoughts on it?

cc: @billsacks

Reply:
If you browse through the new version of `cime_run` (at least on cime master), you'll see that very little is now done directly in `cime_run`: that now mainly farms out to other subroutines. I think the same should be true for glc setup-send. (I'm not sure if that answers your question, though.)